internal/resources/cluster/resource_cluster.go

```diff
@@ -218,15 +227,27 @@ func resourceClusterCreate(ctx context.Context, d *schema.ResourceData, m interf
 
 	var (
 		kubeconfigfile string
+		rawkubeconfig  string
 		k8sclient      k8sClient.Client
 		err            error
 	)
 
 	if _, ok := d.GetOk(attachClusterKey); ok {
-		kubeconfigfile, _ = d.Get(helper.GetFirstElementOf(attachClusterKey, attachClusterKubeConfigKey)).(string)
+		if v, ok := d.GetOk(helper.GetFirstElementOf(attachClusterKey, attachClusterKubeConfigPathKey)); ok {
```

Comment:
can you use switch here to construct the k8s client from file or raw config

Reply:
Done